guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/forms/promote/SelectAssetsToPromoteViewImpl.java

```diff
@@ -101,147 +72,78 @@ public SelectAssetsToPromoteViewImpl() {
 
     }
 
-    public void getOutputMap( @Observes RequestFormParamsEvent event ) {
-
-        Map<String, Object> outputMap = new HashMap<String, Object>();
-
-        int filesToPromoteCount = getFilesToPromoteList().getItemCount();
-        String out_commits = "";
-        for ( int i = 0; i < filesToPromoteCount; i++ ) {
-            String commits = commitsPerFile.get( getFilesToPromoteList().getItemText( i ) );
-            if ( commits == null || commits.length() == 0 || out_commits.contains( commits ) ) {
-                continue;
-            }
-            if ( out_commits.length() > 0 ) {
-                out_commits += ",";
-            }
-            out_commits += commits;
-        }
-
-        outputMap.put( "out_commits", out_commits );
-        outputMap.put( "out_requires_review", getRequiresReviewCheckBox().getValue() );
-        getFormParamsEvent.fire( new GetFormParamsEvent( event.getAction(), outputMap ) );
-
+    @Override
+    public void init( final SelectAssetsToPromotePresenter presenter ) {
+        this.presenter = presenter;
```

Comment:
I know there are versions of MVP implementations around where the view holds a reference to the presenter but I think it's not ideal. The presenter could simply register a callback with the view instead.